server/common/config/app_config.py

```diff
@@ -0,0 +1,171 @@
+import yaml
+from flatten_dict import unflatten
+import os
+from server.common.config.dataset_config import DatasetConfig
+from server.common.config.server_config import ServerConfig
+from server.common.errors import ConfigurationError
+
+
+class AppConfig(object):
+    """AppConfig stores all the configuration for cellxgene.  The configuration is divided into two main parts:
+    server attributes, and dataset attributes.
+    The server_config contains attributes that refer to the server process as a whole.
+    The default_dataset_config refers to attributes that are associated with the features and
+    presentations of a dataset.
+    The dataset config attributes can be overridden depending on the url by which the
+    dataset was accessed.  These are stored in dataroot_config.
+    AppConfig has methods to initialize, modify, and access the configuration.
+    """
+
+    def __init__(self):
+
+        # the default configuration (see default_config.py)
+        ## TODO @madison -- if we always read from the default config (hard coded path) can we set those values as defaults within the config class?
```

Comment:
I meant instead of using the default config file we could just set the default vars. We might need to rewrite the dump function but I think it would simplify quite a bit. 
More of a future optimization...

Reply:
Okay, as long as we maintain the comments that are currently in the config file, and probably also want to maintain the order that the config parameters appear in the file.